packages/sso-express/tests/pkce.test.ts

```diff
@@ -0,0 +1,61 @@
+import type { Request, Response } from "express";
+import { BaseClient, generators } from "openid-client";
+import { mocked } from "ts-jest/utils";
+import { SSOExpressOptions } from "../src";
+import { loginController } from "../src/controllers";
+import { isAuthenticated } from "../src/helpers";
+import { createHash } from "crypto";
+import base64url from "openid-client/lib/helpers/base64url";
+jest.mock("../src/helpers");
+jest.mock("openid-client");
+
+const oidcIssuer = "https://example.com/auth/realms/myRealm";
+const middlewareOptions: SSOExpressOptions = {
+  applicationDomain: ".gov.bc.ca",
+  oidcConfig: {
+    baseUrl: "https://example.com",
+    clientId: "myClient",
+    oidcIssuer,
+    clientSecret: "some-random-secret",
+  },
+  getLandingRoute: jest.fn(),
+};
+
+const client = {
+  metadata: {
+    post_logout_redirect_uris: ["https://example.com/"],
+    redirect_uris: ["https://example.com/auth-callback"],
+  },
+  endSessionUrl: () => "https://oidc-endpoint/logout",
+  refresh: jest.fn(),
+  authorizationUrl: jest.fn(),
+  callbackParams: jest.fn(),
+  callback: jest.fn(),
+} as unknown as BaseClient;
+
+describe("the authCallbackController", () => {
+  it("adds a randomly-generated code_challenge to the session when odicConfig.clientSecret is set ", async () => {
```

Comment:
this could be in `controllers.test.ts`. Our tests file structure matches the source code